encord/client.py

```diff
@@ -504,10 +506,32 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        for _ in range(10):
+            polling_response = self._querier.basic_getter(
+                DatasetDataLongPolling,
+                self._config.resource_id,
+                payload={
+                    "process_hash": process_hash,
+                    "ignore_errors": ignore_errors,
+                },
+            )
+
+            if polling_response["is_done"]:
+                return AddPrivateDataResponse.from_dict(polling_response["response"])
+            else:
+                time.sleep(1)
```

Comment:
why would we need a sleep? I thought this is what the long poll is already doing?

Reply:
+1 - just retry, the sleep is done on the server side ;)

Reply:
sleep is there to guarantee we are not doing very fast requests to BE if something is off and BE returns with close to 0 latency. But it does not have to be there.